indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisor.java

```diff
@@ -761,6 +761,7 @@ public void tryInit()
               catch (InterruptedException e) {
                 stateManager.recordThrowableEvent(e);
                 log.info("SeekableStreamSupervisor[%s] interrupted, exiting", dataSource);
+                Thread.currentThread().interrupt();
```

Comment:
By convention which you use in this PR, `Thread.currentThread().interrupt();` (optionally wrapped in `if (e instanceof InterruptedException)`) is the first statement in the catch block; please move it here above, too. Also in other places in this file `SeekableStreamSupervisor.java`.